libs/kml/serdes_geojson.hpp

```diff
@@ -75,49 +76,37 @@ struct GeoJsonData
 {
   std::string type = "FeatureCollection";
   std::vector<GeoJsonFeature> features;
-  std::map<std::string, std::string> properties;
+  std::optional<std::map<std::string, std::string>> properties;
 };
 
-// Writer and reader
-class GeoJsonWriter
-{
-public:
-  /*DECLARE_EXCEPTION(WriteGeojsonException, RootException);
-
-  explicit GeojsonWriter(Writer & writer)
-    : m_writer(writer)
-  {}
-
-  void Write(FileData const & fileData);
-
-private:
-  Writer & m_writer;*/
-};
+}  // namespace geojson
 
-class GeojsonParser
+// Reader and Writer.
+class DeserializerGeoJson
 {
 public:
-  explicit GeojsonParser(FileData & data) : m_fileData(data) {}
+  DECLARE_EXCEPTION(DeserializeException, RootException);
 
-  bool Parse(std::string_view json_content);
+  explicit DeserializerGeoJson(FileData & fileData) : m_fileData(fileData) {}
+
+  void Deserialize(std::string_view content);
 
 private:
+  bool Parse(std::string_view jsonContent);
   FileData & m_fileData;
 };
 
-}  // namespace geojson
-
-class DeserializerGeoJson
+class GeoJsonWriter
```

Comment:
We have two classes here: `kml::DeserializerGeoJson` and `kml::GeoJsonWriter` doing two opposite things. How to call those classes? Reader/Writer? Parser/Writer? Serializer/Deserializer?

Reply:
Ok, renamed `kml::DeserializerGeoJson` to `kml::GeoJsonReader`.